src/main/java/com/epam/aidial/core/service/PublicationUtil.java

```diff
@@ -50,43 +50,6 @@ public String replaceConversationLinks(String conversationBody, ResourceDescript
         return conversation.toString();
     }
 
-    public String replaceApplicationLinks(String body, ResourceDescription targetResource, boolean preserveReference, Map<String, String> attachmentsMapping) {
```

Comment:
I suppose the new logic is not the same as old.
you lost `preserve` for example

Reply:
By default it is preserved. Publication code generates a new reference.